{"expand":"renderedFields,names,schema,transitions,operations,editmeta,changelog","id":"12428006","self":"https://issues.apache.org/jira/rest/api/latest/issue/12428006","key":"DERBY-4275","fields":{"issuetype":{"self":"https://issues.apache.org/jira/rest/api/2/issuetype/1","id":"1","description":"A problem which impairs or prevents the functions of the product.","iconUrl":"https://issues.apache.org/jira/images/icons/issuetypes/bug.png","name":"Bug","subtask":false},"timespent":null,"project":{"self":"https://issues.apache.org/jira/rest/api/2/project/10594","id":"10594","key":"DERBY","name":"Derby","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/projectavatar?pid=10594&avatarId=10122","24x24":"https://issues.apache.org/jira/secure/projectavatar?size=small&pid=10594&avatarId=10122","16x16":"https://issues.apache.org/jira/secure/projectavatar?size=xsmall&pid=10594&avatarId=10122","32x32":"https://issues.apache.org/jira/secure/projectavatar?size=medium&pid=10594&avatarId=10122"},"projectCategory":{"self":"https://issues.apache.org/jira/rest/api/2/projectCategory/10090","id":"10090","description":"DB related projects","name":"DB"}},"fixVersions":[{"self":"https://issues.apache.org/jira/rest/api/2/version/12313143","id":"12313143","description":"Head of 10.3 branch","name":"10.3.3.1","archived":false,"released":false},{"self":"https://issues.apache.org/jira/rest/api/2/version/12313401","id":"12313401","description":"Head of 10.4 branch","name":"10.4.2.1","archived":false,"released":false},{"self":"https://issues.apache.org/jira/rest/api/2/version/12315436","id":"12315436","description":"Head of 10.5 branch - version bump for DERBY-4835","name":"10.5.3.2","archived":false,"released":false},{"self":"https://issues.apache.org/jira/rest/api/2/version/12315434","id":"12315434","description":"Head of 10.6 branch as of 2014-01-13","name":"10.6.2.4","archived":false,"released":false},{"self":"https://issues.apache.org/jira/rest/api/2/version/12315902","id":"12315902","description":"Head of 10.7 branch as of 2010-11-29","name":"10.7.1.4","archived":false,"released":false},{"self":"https://issues.apache.org/jira/rest/api/2/version/12317968","id":"12317968","description":"second release on the 10.8 branch","name":"10.8.2.2","archived":false,"released":true,"releaseDate":"2011-10-24"},{"self":"https://issues.apache.org/jira/rest/api/2/version/12316344","id":"12316344","description":"First release on the 10.9 branch","name":"10.9.1.0","archived":false,"released":true,"releaseDate":"2012-06-25"}],"aggregatetimespent":null,"resolution":{"self":"https://issues.apache.org/jira/rest/api/2/resolution/1","id":"1","description":"A fix for this issue is checked into the tree and tested.","name":"Fixed"},"customfield_12310220":"2009-07-03 16:12:42.972","customfield_12312322":null,"customfield_12312323":null,"customfield_12310420":"24147","customfield_12310222":"3_*:*_2_*:*_4984046687_*|*_1_*:*_1_*:*_64177683625_*|*_5_*:*_3_*:*_16881016366_*|*_4_*:*_2_*:*_1931962462","customfield_12312320":null,"customfield_12312321":null,"customfield_12312120":null,"customfield_12312121":null,"resolutiondate":"2012-03-30T19:15:00.789+0000","workratio":-1,"customfield_12312328":null,"customfield_12312329":null,"customfield_12312326":null,"customfield_12310300":null,"customfield_12312327":null,"customfield_12312324":null,"customfield_12312325":null,"lastViewed":null,"watches":{"self":"https://issues.apache.org/jira/rest/api/2/issue/DERBY-4275/watchers","watchCount":2,"isWatching":false},"created":"2009-06-16T13:49:51.726+0000","priority":{"self":"https://issues.apache.org/jira/rest/api/2/priority/3","iconUrl":"https://issues.apache.org/jira/images/icons/priorities/major.png","name":"Major","id":"3"},"labels":["derby_triage10_5_2"],"customfield_12312333":null,"customfield_12312334":null,"customfield_12310310":"7.0","customfield_12312331":null,"customfield_12312332":null,"timeestimate":null,"aggregatetimeoriginalestimate":null,"versions":[{"self":"https://issues.apache.org/jira/rest/api/2/version/12313142","id":"12313142","description":"","name":"10.3.3.0","archived":false,"released":true,"releaseDate":"2008-05-12"},{"self":"https://issues.apache.org/jira/rest/api/2/version/12313111","id":"12313111","description":"","name":"10.4.1.3","archived":false,"released":true,"releaseDate":"2008-04-24"},{"self":"https://issues.apache.org/jira/rest/api/2/version/12314117","id":"12314117","description":"Next release after 10.5.2.0 release candidate #1","name":"10.5.3.0","archived":false,"released":true,"releaseDate":"2009-08-21"}],"customfield_12311120":null,"customfield_12312330":null,"issuelinks":[{"id":"12341200","self":"https://issues.apache.org/jira/rest/api/2/issueLink/12341200","type":{"id":"12310000","name":"Duplicate","inward":"is duplicated by","outward":"duplicates","self":"https://issues.apache.org/jira/rest/api/2/issueLinkType/12310000"},"inwardIssue":{"id":"12515207","key":"DERBY-5351","self":"https://issues.apache.org/jira/rest/api/2/issue/12515207","fields":{"summary":"SYSCS_UTIL.SYSCS_COMPRESS_TABLE causes concurrent transactions to fail","status":{"self":"https://issues.apache.org/jira/rest/api/2/status/6","description":"The issue is considered finished, the resolution is correct. Issues which are closed can be reopened.","iconUrl":"https://issues.apache.org/jira/images/icons/statuses/closed.png","name":"Closed","id":"6","statusCategory":{"self":"https://issues.apache.org/jira/rest/api/2/statuscategory/3","id":3,"key":"done","colorName":"green","name":"Complete"}},"priority":{"self":"https://issues.apache.org/jira/rest/api/2/priority/3","iconUrl":"https://issues.apache.org/jira/images/icons/priorities/major.png","name":"Major","id":"3"},"issuetype":{"self":"https://issues.apache.org/jira/rest/api/2/issuetype/1","id":"1","description":"A problem which impairs or prevents the functions of the product.","iconUrl":"https://issues.apache.org/jira/images/icons/issuetypes/bug.png","name":"Bug","subtask":false}}}},{"id":"12339386","self":"https://issues.apache.org/jira/rest/api/2/issueLink/12339386","type":{"id":"10030","name":"Reference","inward":"is related to","outward":"relates to","self":"https://issues.apache.org/jira/rest/api/2/issueLinkType/10030"},"outwardIssue":{"id":"12396485","key":"DERBY-3683","self":"https://issues.apache.org/jira/rest/api/2/issue/12396485","fields":{"summary":"SYSCS_COMPRESS_TABLE gets deadlock while executing concurrently with other user threads","status":{"self":"https://issues.apache.org/jira/rest/api/2/status/1","description":"The issue is open and ready for the assignee to start work on it.","iconUrl":"https://issues.apache.org/jira/images/icons/statuses/open.png","name":"Open","id":"1","statusCategory":{"self":"https://issues.apache.org/jira/rest/api/2/statuscategory/2","id":2,"key":"new","colorName":"blue-gray","name":"New"}},"priority":{"self":"https://issues.apache.org/jira/rest/api/2/priority/4","iconUrl":"https://issues.apache.org/jira/images/icons/priorities/minor.png","name":"Minor","id":"4"},"issuetype":{"self":"https://issues.apache.org/jira/rest/api/2/issuetype/1","id":"1","description":"A problem which impairs or prevents the functions of the product.","iconUrl":"https://issues.apache.org/jira/images/icons/issuetypes/bug.png","name":"Bug","subtask":false}}}},{"id":"12330728","self":"https://issues.apache.org/jira/rest/api/2/issueLink/12330728","type":{"id":"10030","name":"Reference","inward":"is related to","outward":"relates to","self":"https://issues.apache.org/jira/rest/api/2/issueLinkType/10030"},"outwardIssue":{"id":"12318206","key":"DERBY-637","self":"https://issues.apache.org/jira/rest/api/2/issue/12318206","fields":{"summary":"Conglomerate does not exist after inserting large data  volume","status":{"self":"https://issues.apache.org/jira/rest/api/2/status/6","description":"The issue is considered finished, the resolution is correct. Issues which are closed can be reopened.","iconUrl":"https://issues.apache.org/jira/images/icons/statuses/closed.png","name":"Closed","id":"6","statusCategory":{"self":"https://issues.apache.org/jira/rest/api/2/statuscategory/3","id":3,"key":"done","colorName":"green","name":"Complete"}},"priority":{"self":"https://issues.apache.org/jira/rest/api/2/priority/3","iconUrl":"https://issues.apache.org/jira/images/icons/priorities/major.png","name":"Major","id":"3"},"issuetype":{"self":"https://issues.apache.org/jira/rest/api/2/issuetype/1","id":"1","description":"A problem which impairs or prevents the functions of the product.","iconUrl":"https://issues.apache.org/jira/images/icons/issuetypes/bug.png","name":"Bug","subtask":false}}}},{"id":"12341310","self":"https://issues.apache.org/jira/rest/api/2/issueLink/12341310","type":{"id":"10030","name":"Reference","inward":"is related to","outward":"relates to","self":"https://issues.apache.org/jira/rest/api/2/issueLinkType/10030"},"inwardIssue":{"id":"12515751","key":"DERBY-5358","self":"https://issues.apache.org/jira/rest/api/2/issue/12515751","fields":{"summary":"SYSCS_COMPRESS_TABLE failed with conglomerate not found exception","status":{"self":"https://issues.apache.org/jira/rest/api/2/status/6","description":"The issue is considered finished, the resolution is correct. Issues which are closed can be reopened.","iconUrl":"https://issues.apache.org/jira/images/icons/statuses/closed.png","name":"Closed","id":"6","statusCategory":{"self":"https://issues.apache.org/jira/rest/api/2/statuscategory/3","id":3,"key":"done","colorName":"green","name":"Complete"}},"priority":{"self":"https://issues.apache.org/jira/rest/api/2/priority/3","iconUrl":"https://issues.apache.org/jira/images/icons/priorities/major.png","name":"Major","id":"3"},"issuetype":{"self":"https://issues.apache.org/jira/rest/api/2/issuetype/1","id":"1","description":"A problem which impairs or prevents the functions of the product.","iconUrl":"https://issues.apache.org/jira/images/icons/issuetypes/bug.png","name":"Bug","subtask":false}}}},{"id":"12343029","self":"https://issues.apache.org/jira/rest/api/2/issueLink/12343029","type":{"id":"10030","name":"Reference","inward":"is related to","outward":"relates to","self":"https://issues.apache.org/jira/rest/api/2/issueLinkType/10030"},"inwardIssue":{"id":"12522398","key":"DERBY-5406","self":"https://issues.apache.org/jira/rest/api/2/issue/12522398","fields":{"summary":"Intermittent failures in CompressTableTest and TruncateTableTest","status":{"self":"https://issues.apache.org/jira/rest/api/2/status/6","description":"The issue is considered finished, the resolution is correct. Issues which are closed can be reopened.","iconUrl":"https://issues.apache.org/jira/images/icons/statuses/closed.png","name":"Closed","id":"6","statusCategory":{"self":"https://issues.apache.org/jira/rest/api/2/statuscategory/3","id":3,"key":"done","colorName":"green","name":"Complete"}},"priority":{"self":"https://issues.apache.org/jira/rest/api/2/priority/3","iconUrl":"https://issues.apache.org/jira/images/icons/priorities/major.png","name":"Major","id":"3"},"issuetype":{"self":"https://issues.apache.org/jira/rest/api/2/issuetype/1","id":"1","description":"A problem which impairs or prevents the functions of the product.","iconUrl":"https://issues.apache.org/jira/images/icons/issuetypes/bug.png","name":"Bug","subtask":false}}}},{"id":"12349275","self":"https://issues.apache.org/jira/rest/api/2/issueLink/12349275","type":{"id":"12310040","name":"Required","inward":"is required by","outward":"requires","self":"https://issues.apache.org/jira/rest/api/2/issueLinkType/12310040"},"inwardIssue":{"id":"12546430","key":"DERBY-5654","self":"https://issues.apache.org/jira/rest/api/2/issue/12546430","fields":{"summary":"10.5 Backport issue (2012)","status":{"self":"https://issues.apache.org/jira/rest/api/2/status/6","description":"The issue is considered finished, the resolution is correct. Issues which are closed can be reopened.","iconUrl":"https://issues.apache.org/jira/images/icons/statuses/closed.png","name":"Closed","id":"6","statusCategory":{"self":"https://issues.apache.org/jira/rest/api/2/statuscategory/3","id":3,"key":"done","colorName":"green","name":"Complete"}},"priority":{"self":"https://issues.apache.org/jira/rest/api/2/priority/4","iconUrl":"https://issues.apache.org/jira/images/icons/priorities/minor.png","name":"Minor","id":"4"},"issuetype":{"self":"https://issues.apache.org/jira/rest/api/2/issuetype/3","id":"3","description":"A task that needs to be done.","iconUrl":"https://issues.apache.org/jira/images/icons/issuetypes/task.png","name":"Task","subtask":false}}}}],"customfield_12312339":null,"assignee":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"customfield_12312337":null,"customfield_12312338":null,"updated":"2013-06-17T09:19:44.238+0000","customfield_12312335":null,"customfield_12311720":null,"customfield_12312336":null,"status":{"self":"https://issues.apache.org/jira/rest/api/2/status/6","description":"The issue is considered finished, the resolution is correct. Issues which are closed can be reopened.","iconUrl":"https://issues.apache.org/jira/images/icons/statuses/closed.png","name":"Closed","id":"6","statusCategory":{"self":"https://issues.apache.org/jira/rest/api/2/statuscategory/3","id":3,"key":"done","colorName":"green","name":"Complete"}},"components":[{"self":"https://issues.apache.org/jira/rest/api/2/component/11412","id":"11412","name":"Store"}],"timeoriginalestimate":null,"description":"Query executions (SELECT and/or UPDATE) fail with serious exceptions while the table is being compressed using SYSCS_UTIL.SYSCS_COMPRESS_ TABLE. The compression eventually finishes normally, but the queries keep failing with the same error until the database is rebooted. More information about this can be found on the Derby mailing list at http://www.nabble.com/Issue-with-SYSCS_UTIL.SYSCS_COMPRESS_-TABLE-td23892893.html. The exception stacktrace is below: \r\n\r\nCaused by: java.sql.SQLException: The conglomerate (71,409) requested does not exist.\r\n            at org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(Unknown Source)\r\n            at org.apache.derby.impl.jdbc.Util.generateCsSQLException(Unknown Source)\r\n            at org.apache.derby.impl.jdbc.TransactionResourceImpl.wrapInSQLException(Unknown Source)\r\n            at org.apache.derby.impl.jdbc.TransactionResourceImpl.handleException(Unknown Source)\r\n            at org.apache.derby.impl.jdbc.EmbedConnection.handleException(Unknown Source)\r\n            at org.apache.derby.impl.jdbc.ConnectionChild.handleException(Unknown Source)\r\n            at org.apache.derby.impl.jdbc.EmbedStatement.executeStatement(Unknown Source)\r\n            at org.apache.derby.impl.jdbc.EmbedPreparedStatement.executeStatement(Unknown Source)\r\n            at org.apache.derby.impl.jdbc.EmbedPreparedStatement.executeQuery(Unknown Source)\r\n            at org.apache.tomcat.dbcp.dbcp.DelegatingPreparedStatement.executeQuery(DelegatingPreparedStatement.java:93)\r\n            ... 25 more\r\n        Caused by: ERROR XSAI2: The conglomerate (71,409) requested does not exist.\r\n            at org.apache.derby.iapi.error.StandardException.newException(Unknown Source)\r\n            at org.apache.derby.impl.store.access.btree.index.B2IFactory.readConglomerate(Unknown Source)\r\n            at org.apache.derby.impl.store.access.RAMAccessManager.conglomCacheFind(Unknown Source)\r\n            at org.apache.derby.impl.store.access.RAMTransaction.findExistingConglomerate(Unknown Source)\r\n            at org.apache.derby.impl.store.access.RAMTransaction.openScan(Unknown Source)\r\n            at org.apache.derby.impl.store.access.BackingStoreHashTableFromScan.<init>(Unknown Source)\r\n            at org.apache.derby.impl.store.access.RAMTransaction.createBackingStoreHashtableFromScan(Unknown Source)\r\n            at org.apache.derby.impl.sql.execute.HashScanResultSet.openCore(Unknown Source)\r\n            at org.apache.derby.impl.sql.execute.JoinResultSet.openRight(Unknown Source)\r\n            at org.apache.derby.impl.sql.execute.JoinResultSet.openCore(Unknown Source)\r\n            at org.apache.derby.impl.sql.execute.JoinResultSet.openCore(Unknown Source)\r\n            at org.apache.derby.impl.sql.execute.ProjectRestrictResultSet.openCore(Unknown Source)\r\n            at org.apache.derby.impl.sql.execute.UnionResultSet.getNextRowCore(Unknown Source)\r\n            at org.apache.derby.impl.sql.execute.SortResultSet.getRowFromResultSet(Unknown Source)\r\n            at org.apache.derby.impl.sql.execute.SortResultSet.getNextRowFromRS(Unknown Source)\r\n            at org.apache.derby.impl.sql.execute.SortResultSet.loadSorter(Unknown Source)\r\n            at org.apache.derby.impl.sql.execute.SortResultSet.openCore(Unknown Source)\r\n            at org.apache.derby.impl.sql.execute.SortResultSet.openCore(Unknown Source)\r\n            at org.apache.derby.impl.sql.execute.BasicNoPutResultSetImpl.open(Unknown Source)\r\n            at org.apache.derby.impl.sql.GenericPreparedStatement.execute(Unknown Source) ","customfield_10010":null,"timetracking":{},"customfield_12312026":null,"customfield_12310200":[{"self":"https://issues.apache.org/jira/rest/api/2/customFieldOption/10365","value":"Crash","id":"10365"},{"self":"https://issues.apache.org/jira/rest/api/2/customFieldOption/10421","value":"Seen in production","id":"10421"}],"customfield_12312023":null,"customfield_12312024":null,"customfield_12312340":null,"aggregatetimeestimate":null,"customfield_12312022":null,"customfield_12312341":null,"customfield_12312220":null,"customfield_12310921":null,"customfield_12310920":"35842","summary":"Query executions fail when compressing a table using SYSCS_UTIL.SYSCS_COMPRESS_TABLE","creator":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=psai","name":"psai","emailAddress":"sai dot pullabhotla at jmethods dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Sai Pullabhotla","active":true},"subtasks":[],"customfield_12310090":[{"self":"https://issues.apache.org/jira/rest/api/2/customFieldOption/10422","value":"High Value Fix","id":"10422"},{"self":"https://issues.apache.org/jira/rest/api/2/customFieldOption/10424","value":"Repro attached","id":"10424"}],"reporter":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=psai","name":"psai","emailAddress":"sai dot pullabhotla at jmethods dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Sai Pullabhotla","active":true},"customfield_12310291":null,"customfield_12310290":null,"aggregateprogress":{"progress":0,"total":0},"environment":null,"customfield_12311020":null,"customfield_12310050":{"self":"https://issues.apache.org/jira/rest/api/2/customFieldOption/10052","value":"Normal","id":"10052"},"duedate":null,"customfield_12310250":null,"progress":{"progress":0,"total":0},"comment":{"startAt":0,"maxResults":27,"total":27,"comments":[{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/12720137","id":"12720137","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=psai","name":"psai","emailAddress":"sai dot pullabhotla at jmethods dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Sai Pullabhotla","active":true},"body":"The attached files can be used to reproduce this reported bug. \r\n\r\nCompressDBTest1.java - tries to read a table while it is being compressed. The select statement errors out with the following error:\r\njava.sql.SQLException: Container 2,192 not found.\r\nSubsequent SELECTs with a brand new connection also fail with a little bit different error:\r\nThe conglomerate (2,192) requested does not exist.\r\n\r\nCompressDBTest2.java - tries to insert data into a table while it is being compressed. The insert errors out eventually with the following error:\r\nA lock could not be obtained due to a deadlock, cycle of locks and waiters is:\r\nLock : TABLE, TEST, Tablelock\r\n  Waiting XID : {236439, IX} , APP, insert into test values(?, ?, ?, ?, ?)\r\n  Granted XID : {234342, X}\r\nLock : ROW, SYSCONGLOMERATES, (5,14)\r\n  Waiting XID : {234342, X} , APP, alter table \"APP\".\"TEST\" compress sequential\r\n  Granted XID : {234342, S} , {236439, S}\r\n. The selected victim is XID : 236439.\r\n\r\nWhat the test classes do:\r\n\r\nBoth classes first create a database named test under the current working directory if a directory named test does not already exist. Then a table named test is created and populated with 100,000 random records. Two threads are started then with the first one repeatedly compressing the test table and the second one repeatedly executing a DML statement. If the DML statement errors out, the compression thread will be stopped. Finally the database is shutdown. The standard output of the program goes to stdout.log and standard error goes to stderr.log in the current working directory.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=psai","name":"psai","emailAddress":"sai dot pullabhotla at jmethods dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Sai Pullabhotla","active":true},"created":"2009-06-16T13:55:05.325+0000","updated":"2009-06-16T13:55:05.325+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/12727024","id":"12727024","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=dagw","name":"dagw","emailAddress":"dag dot wanvik at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=dagw&avatarId=16789","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=dagw&avatarId=16789","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=dagw&avatarId=16789","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=dagw&avatarId=16789"},"displayName":"Dag H. Wanvik","active":true},"body":"Thanks for the repro, Sai! I was able to reproduce with Derby 10.4.1.3.\r\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=dagw","name":"dagw","emailAddress":"dag dot wanvik at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=dagw&avatarId=16789","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=dagw&avatarId=16789","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=dagw&avatarId=16789","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=dagw&avatarId=16789"},"displayName":"Dag H. Wanvik","active":true},"created":"2009-07-03T16:12:42.972+0000","updated":"2009-07-03T16:12:42.972+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/12727026","id":"12727026","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=dagw","name":"dagw","emailAddress":"dag dot wanvik at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=dagw&avatarId=16789","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=dagw&avatarId=16789","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=dagw&avatarId=16789","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=dagw&avatarId=16789"},"displayName":"Dag H. Wanvik","active":true},"body":"Triaged for 10.5.2, checking  \"repro attached\".","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=dagw","name":"dagw","emailAddress":"dag dot wanvik at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=dagw&avatarId=16789","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=dagw&avatarId=16789","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=dagw&avatarId=16789","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=dagw&avatarId=16789"},"displayName":"Dag H. Wanvik","active":true},"created":"2009-07-03T16:13:49.160+0000","updated":"2009-07-03T16:13:49.160+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/12844012","id":"12844012","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"body":"Some of the discussion on DERBY-637 may be relevant for this issue as well.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"created":"2010-03-11T10:54:13.347+0000","updated":"2010-03-11T10:54:13.347+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/12844036","id":"12844036","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=brettw","name":"brettw","emailAddress":"brett dot wooldridge at gmail dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Brett Wooldridge","active":true},"body":"Throwing my vote in for this bug.  We were planning on enabling online compression, statistics regeneration, and backup in an upcoming release of our software.\r\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=brettw","name":"brettw","emailAddress":"brett dot wooldridge at gmail dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Brett Wooldridge","active":true},"created":"2010-03-11T11:43:35.180+0000","updated":"2010-03-11T11:43:35.180+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/13042119","id":"13042119","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"body":"The problem shown by CompressDBTest2.java looks like DERBY-3683.\r\n\r\nThe \"conglomerate does not exist\" errors in CompressDBTest1.java seem to happen because AlterTableConstantAction.compressTable() invalidates the prepared statements before it compresses the table. Executing statements may then try to recompile before the table has been compressed, and they continue using the old conglomerate since they're not invalidated again.\r\n\r\nI tried to delay the invalidation until the new compressed conglomerates had been created and the system tables were updated (see attached patch), and that made the CompressDBTest1 repro complete with no failures.\r\n\r\nThere is a comment saying that the early invalidation was intentional to work around a bug. Looking at an earlier version of the comment, it referred to a bug 3653 where some statements didn't recompile if the invalidation happened later. I don't know what kind of statements had that problem, though. I'll run the regression tests to see if they show anything. (Truncate table has a similar comment and would probably need a similar fix.)\r\n\r\nThere's always the possibility to invalidate statements both before and after compressing the table, but it really should be enough to do it once.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"created":"2011-06-01T11:39:36.300+0000","updated":"2011-06-01T11:39:36.300+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/13046502","id":"13046502","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"body":"Running the regression test suites didn't reveal any problems with the patch. I'll see if I can construct a case that runs into problem mentioned in the comment:\r\n\r\n\t\t// invalidate any prepared statements that\r\n\t\t// depended on this table (including this one)\r\n\t\t// bug 3653 has threads that start up and block on our lock, but do\r\n\t\t// not see they have to recompile their plan.    We now invalidate earlier\r\n\t\t// however they still might recompile using the old conglomerate id before we\r\n\t\t// commit our DD changes.\r\n\r\nThe scenario described in the last sentence of that comment, is exactly what's happening when we see the bug.\r\n\r\nI'm not sure why other threads would fail to see that they need to recompile if the invalidation happens later. They may fail because the conglomerate doesn't exist anymore, but the result set classes have code that checks if the statement is still valid if an error happens, and recompiles and reexecutes if it's not valid, so that case is supposed to be handled. But maybe that code was added after bug 3653 had been fixed, I don't know...","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"created":"2011-06-09T12:28:52.007+0000","updated":"2011-06-09T12:28:52.007+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/13054327","id":"13054327","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"body":"During SYSCS_COMPRESS_TABLE we invalidate dependent statements three times, and all of them happen before any changes have been made to the data dictionary. With the patch, this is changed to two invalidations before the DD changes and one after the DD is updated.\r\n\r\nThe first two invalidations are done by the general alter table machinery. The compress table implementation was moved into alter table in 10.5 (DERBY-1062). It may be that the earlier invalidation was needed before that change, but now that we have the alter table machinery doing early invalidation in any case, the bug mentioned in the comments (bug 3653) is probably handled even if we move the compress-specific invalidation after the data dictionary changes.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"created":"2011-06-24T09:37:31.953+0000","updated":"2011-06-24T09:37:31.953+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/13057730","id":"13057730","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"body":"Attached is a patch which adds JUnit test cases that reproduce this bug with compress table and truncate table. The patch also moves the third invalidation on compress/truncate to a point where the new conglomerate information has been written to the system tables, so that the statements get recompiled using the new conglomerates.\r\n\r\nAll the regression tests ran cleanly with the patch.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"created":"2011-06-30T09:57:56.128+0000","updated":"2011-06-30T09:57:56.128+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/13059351","id":"13059351","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"body":"Committed revision 1142583.\r\n\r\nI also plan to backport the fix to the 10.8 branch.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"created":"2011-07-04T07:54:17.583+0000","updated":"2011-07-04T07:54:17.583+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/13060492","id":"13060492","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"body":"Merged to 10.8. Committed revision 1143356.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"created":"2011-07-06T11:42:03.751+0000","updated":"2011-07-06T11:42:03.751+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/13065256","id":"13065256","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kristwaa","name":"kristwaa","emailAddress":"kristwaa at apache dot org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=kristwaa&avatarId=14736","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=kristwaa&avatarId=14736","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=kristwaa&avatarId=14736","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=kristwaa&avatarId=14736"},"displayName":"Kristian Waagan","active":true},"body":"I'm still seeing this on a Linux machine:\r\nFedora 15\r\nLinux work2 2.6.38.8-32.fc15.x86_64 #1 SMP Mon Jun 13 19:49:05 UTC 2011 x86_64 x86_64 x86_64 GNU/Linux\r\nIntel(R) Core(TM)2 Duo CPU     E8500  @ 3.16GHz\r\njava version \"1.6.0_22\"\r\nOpenJDK Runtime Environment (IcedTea6 1.10.2) (fedora-58.1.10.2.fc15-x86_64)\r\nOpenJDK 64-Bit Server VM (build 20.0-b11, mixed mode)\r\n\r\nTo reproduce I ran two instances of the test in a tight loop in the shell. I've also seen it reproduce when I've used the parallel test runner.\r\n\r\nLast minute update, it also reproduces on Solaris 11:\r\njava version \"1.6.0_24\"\r\nJava(TM) SE Runtime Environment (build 1.6.0_24-b07)\r\nJava HotSpot(TM) Server VM (build 19.1-b02, mixed mode)\r\n\r\n\r\nTime: 5.035\r\nThere was 1 failure:\r\n1) testConcurrentInvalidation(org.apache.derbyTesting.functionTests.tests.lang.CompressTableTest)junit.framework.AssertionFailedError: Helper thread failed\r\n\tat org.apache.derbyTesting.junit.BaseTestCase.fail(BaseTestCase.java:801)\r\n\tat org.apache.derbyTesting.functionTests.tests.lang.CompressTableTest.testConcurrentInvalidation(CompressTableTest.java:121)\r\n\tat sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\r\n\tat sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)\r\n\tat sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\r\n\tat org.apache.derbyTesting.junit.BaseTestCase.runBare(BaseTestCase.java:112)\r\n\tat junit.extensions.TestDecorator.basicRun(TestDecorator.java:24)\r\n\tat junit.extensions.TestSetup$1.protect(TestSetup.java:21)\r\n\tat junit.extensions.TestSetup.run(TestSetup.java:25)\r\n\tat org.apache.derbyTesting.junit.BaseTestSetup.run(BaseTestSetup.java:57)\r\nCaused by: java.sql.SQLException: The conglomerate (2,720) requested does not exist.\r\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:98)\r\n\tat org.apache.derby.impl.jdbc.Util.generateCsSQLException(Util.java:256)\r\n\tat org.apache.derby.impl.jdbc.TransactionResourceImpl.wrapInSQLException(TransactionResourceImpl.java:400)\r\n\tat org.apache.derby.impl.jdbc.TransactionResourceImpl.handleException(TransactionResourceImpl.java:348)\r\n\tat org.apache.derby.impl.jdbc.EmbedConnection.handleException(EmbedConnection.java:2290)\r\n\tat org.apache.derby.impl.jdbc.ConnectionChild.handleException(ConnectionChild.java:82)\r\n\tat org.apache.derby.impl.jdbc.EmbedStatement.executeStatement(EmbedStatement.java:1334)\r\n\tat org.apache.derby.impl.jdbc.EmbedPreparedStatement.executeStatement(EmbedPreparedStatement.java:1686)\r\n\tat org.apache.derby.impl.jdbc.EmbedPreparedStatement.executeQuery(EmbedPreparedStatement.java:284)\r\n\tat org.apache.derbyTesting.functionTests.tests.lang.CompressTableTest$1.run(CompressTableTest.java:92)\r\nCaused by: java.sql.SQLException: The conglomerate (2,720) requested does not exist.\r\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(SQLExceptionFactory.java:45)\r\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory40.wrapArgsForTransportAcrossDRDA(SQLExceptionFactory40.java:122)\r\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:71)\r\n\t... 9 more\r\nCaused by: ERROR XSAI2: The conglomerate (2,720) requested does not exist.\r\n\tat org.apache.derby.iapi.error.StandardException.newException(StandardException.java:278)\r\n\tat org.apache.derby.impl.store.access.heap.HeapConglomerateFactory.readConglomerate(HeapConglomerateFactory.java:254)\r\n\tat org.apache.derby.impl.store.access.RAMAccessManager.conglomCacheFind(RAMAccessManager.java:482)\r\n\tat org.apache.derby.impl.store.access.RAMTransaction.findExistingConglomerate(RAMTransaction.java:394)\r\n\tat org.apache.derby.impl.store.access.RAMTransaction.getDynamicCompiledConglomInfo(RAMTransaction.java:692)\r\n\tat org.apache.derby.impl.sql.execute.TableScanResultSet.openCore(TableScanResultSet.java:245)\r\n\tat org.apache.derby.impl.sql.execute.BulkTableScanResultSet.openCore(BulkTableScanResultSet.java:248)\r\n\tat org.apache.derby.impl.sql.execute.BasicNoPutResultSetImpl.open(BasicNoPutResultSetImpl.java:255)\r\n\tat org.apache.derby.impl.sql.GenericPreparedStatement.executeStmt(GenericPreparedStatement.java:436)\r\n\tat org.apache.derby.impl.sql.GenericPreparedStatement.execute(GenericPreparedStatement.java:317)\r\n\tat org.apache.derby.impl.jdbc.EmbedStatement.executeStatement(EmbedStatement.java:1242)\r\n\t... 3 more\r\n\r\nFAILURES!!!\r\nTests run: 2,  Failures: 1,  Errors: 0\r\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kristwaa","name":"kristwaa","emailAddress":"kristwaa at apache dot org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=kristwaa&avatarId=14736","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=kristwaa&avatarId=14736","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=kristwaa&avatarId=14736","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=kristwaa&avatarId=14736"},"displayName":"Kristian Waagan","active":true},"created":"2011-07-14T13:37:00.273+0000","updated":"2011-07-14T13:37:00.273+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/13065258","id":"13065258","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kristwaa","name":"kristwaa","emailAddress":"kristwaa at apache dot org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=kristwaa&avatarId=14736","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=kristwaa&avatarId=14736","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=kristwaa&avatarId=14736","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=kristwaa&avatarId=14736"},"displayName":"Kristian Waagan","active":true},"body":"Forgot to mention that I used a sane build and ran from classes.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kristwaa","name":"kristwaa","emailAddress":"kristwaa at apache dot org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=kristwaa&avatarId=14736","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=kristwaa&avatarId=14736","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=kristwaa&avatarId=14736","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=kristwaa&avatarId=14736"},"displayName":"Kristian Waagan","active":true},"created":"2011-07-14T13:37:59.757+0000","updated":"2011-07-14T13:37:59.757+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/13068606","id":"13068606","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"body":"I've also seen this happen frequently in my test runs on phoneME running inside a VirtualBox instance, so it seems that the fix was incomplete. I'll see if I can get it reproduced in a saner environment and do some debugging.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"created":"2011-07-20T20:21:04.931+0000","updated":"2011-07-20T20:21:04.931+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/13071739","id":"13071739","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"body":"I managed to reproduce it in my environment and have been running with an instrumented build to try to figure out what's going on. It looks to me as if the problem is that GenericPreparedStatement.makeInvalid() does nothing if the statement is in the process of being recompiled. That may be fine if the recompilation has just started, but if it's in the middle of the compilation, it may end up producing an execution plan that only works with the old conglomerate.\r\n\r\nI've experimented with the attached patch (invalidation-during-compilation.diff). That patch makes GPS.makeInvalid() set a flag if it attempted an invalidation during compilation, and GPS.executeStmt() checks the flag after compilation and requests recompilation if necessary.\r\n\r\nThis makes the failure happen less frequently. It appears to completely fix the situations where makeInvalid() used to swallow the invalidation request and the recompilation successfully completed using stale meta-data, which caused the execution of the plan to fail with a conglomerate not found exception. The stack trace Kristian posted was such a case.\r\n\r\nHowever, there's a variation that's not properly handled by the patch. If the conglomerate not found exception is thrown by the in-progress recompilation itself, because the conglomerate disappeared under its feet, something goes wrong. The patch does make it ignore the exception and recompile once more, the recompilation succeeds, and so does the execution of the plan. But when the transaction in which this happened is committed, the following exception is thrown:\r\n\r\nERROR X0Y66: Cannot issue commit in a nested connection when there is a pending operation in the parent connection.\r\n\r\nProbably the context stack is in a bad state after the conglomerate not found exception and needs to be restored somehow, but I haven't dug into that yet.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"created":"2011-07-27T14:02:07.649+0000","updated":"2011-07-27T14:02:07.649+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/13071779","id":"13071779","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"body":"When I combined the invalidation-during-compilation.diff patch with the patch for DERBY-5161 (which had to be backed out because of DERBY-5280), I didn't see the error on commit anymore. I've been testing the patches by running three parallel processes of the attached Java program D4275.java.\r\n\r\nHowever, after 12 minutes, one of the test processes failed with a NullPointerException:\r\n\r\njava.lang.NullPointerException\r\n\tat org.apache.derby.impl.sql.compile.FromBaseTable.getStoreCostController(FromBaseTable.java:4515)\r\n\tat org.apache.derby.impl.sql.compile.FromBaseTable.getBaseCostController(FromBaseTable.java:4521)\r\n\tat org.apache.derby.impl.sql.compile.FromBaseTable.estimateCost(FromBaseTable.java:1428)\r\n\tat org.apache.derby.impl.sql.compile.OptimizerImpl.estimateTotalCost(OptimizerImpl.java:2626)\r\n\tat org.apache.derby.impl.sql.compile.OptimizerImpl.costBasedCostOptimizable(OptimizerImpl.java:2172)\r\n\tat org.apache.derby.impl.sql.compile.OptimizerImpl.costOptimizable(OptimizerImpl.java:1985)\r\n\tat org.apache.derby.impl.sql.compile.FromBaseTable.optimizeIt(FromBaseTable.java:524)\r\n\tat org.apache.derby.impl.sql.compile.ProjectRestrictNode.optimizeIt(ProjectRestrictNode.java:316)\r\n\tat org.apache.derby.impl.sql.compile.OptimizerImpl.costPermutation(OptimizerImpl.java:1939)\r\n\tat org.apache.derby.impl.sql.compile.SelectNode.optimize(SelectNode.java:1913)\r\n\tat org.apache.derby.impl.sql.compile.DMLStatementNode.optimizeStatement(DMLStatementNode.java:315)\r\n\tat org.apache.derby.impl.sql.compile.CursorNode.optimizeStatement(CursorNode.java:587)\r\n\tat org.apache.derby.impl.sql.GenericStatement.prepMinion(GenericStatement.java:384)\r\n\tat org.apache.derby.impl.sql.GenericStatement.prepare(GenericStatement.java:85)\r\n\tat org.apache.derby.impl.sql.GenericPreparedStatement.rePrepare(GenericPreparedStatement.java:231)\r\n\tat org.apache.derby.impl.sql.GenericPreparedStatement.executeStmt(GenericPreparedStatement.java:412)\r\n\tat org.apache.derby.impl.sql.GenericPreparedStatement.execute(GenericPreparedStatement.java:319)\r\n\tat org.apache.derby.impl.jdbc.EmbedStatement.executeStatement(EmbedStatement.java:1242)\r\n\tat org.apache.derby.impl.jdbc.EmbedPreparedStatement.executeStatement(EmbedPreparedStatement.java:1686)\r\n\tat org.apache.derby.impl.jdbc.EmbedPreparedStatement.executeQuery(EmbedPreparedStatement.java:284)\r\n\tat D4275$1.run0(D4275.java:35)\r\n\tat D4275$1.run(D4275.java:23)\r\n\r\nI don't know if this is caused by the fix, or if it's some other existing problem that popped up by coincidence. The other two test processes didn't show any problems, and I terminated them after 30 minutes. Before I included the DERBY-5161 fix, the test would typically fail in less than a minute.\r\n\r\nIf DERBY-5161 + invalidation-during-compilation.diff (+ whatever extra is needed to fix the NPE) is the right solution for this issue, we'll need to find out how to reintroduce the DERBY-5161 fix without reintroducing DERBY-5280.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"created":"2011-07-27T14:57:15.398+0000","updated":"2011-07-27T14:57:15.398+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/13071965","id":"13071965","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=dagw","name":"dagw","emailAddress":"dag dot wanvik at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=dagw&avatarId=16789","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=dagw&avatarId=16789","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=dagw&avatarId=16789","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=dagw&avatarId=16789"},"displayName":"Dag H. Wanvik","active":true},"body":"Interesting find. Sounds like reintroducing a (better) fix for DERBY-5161 might be a useful first step.. The cleanup/context  popping machinery is complex, though, so I'm just guessing. Thanks for digging into this.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=dagw","name":"dagw","emailAddress":"dag dot wanvik at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=dagw&avatarId=16789","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=dagw&avatarId=16789","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=dagw&avatarId=16789","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=dagw&avatarId=16789"},"displayName":"Dag H. Wanvik","active":true},"created":"2011-07-27T20:08:44.628+0000","updated":"2011-07-27T20:08:44.628+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/13072718","id":"13072718","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"body":"I had my test case running overnight to find out exactly what was causing the NPE (there were two references that could be null on the line number reported). I saw the NPE twice (after two and three hours). It looks like the NPE happens because a conglomerate is not found and a conglomerate descriptor ends up as null. So it sounds like a case where a conglomerate not found exception would be in order. If a StandardException is thrown instead of a NullPointerException, the code knows how to recover (recompile and retry).\r\n\r\nThe attached patch (npe.diff) checks if a conglomerate descriptor was found, and raises a StandardException if not.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"created":"2011-07-29T07:26:51.511+0000","updated":"2011-07-29T07:26:51.511+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/13072765","id":"13072765","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"body":"All the regression tests ran cleanly with the npe.diff patch. The patch doesn't have a regression test case, as I don't have a reliable way to reproduce the NPE in reasonable time. I do think however that it can happen in the existing tests CompressTableTest and TruncateTableTest when the timing is right.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"created":"2011-07-29T10:14:02.869+0000","updated":"2011-07-29T10:14:02.869+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/13089365","id":"13089365","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"body":"Committed npe.diff to trunk with revision 1160597.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"created":"2011-08-23T09:36:38.770+0000","updated":"2011-08-23T09:36:38.770+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/13095934","id":"13095934","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"body":"Merged npe.diff to 10.8 and committed revision 1164499.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"created":"2011-09-02T12:07:16.126+0000","updated":"2011-09-02T12:07:16.126+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/13101101","id":"13101101","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"body":"I'm marking the issue as resolved again so that it turns up in the 10.8.2 release notes, since the original fix did solve some of the problems in this area.\r\n\r\nI've filed DERBY-5406 to track the remaining problems revealed by the regression test cases for this bug.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"created":"2011-09-09T10:04:30.677+0000","updated":"2011-09-09T10:04:30.677+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/13229393","id":"13229393","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kmarsden","name":"kmarsden","emailAddress":"kmarsdenderby at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Kathey Marsden","active":true},"body":"Reopen for 10.5 backport consideration. If working on the backport for this issue. Temporarily assign yourself and add a comment that you are working on it. When finished, reresolve with the new fix versions or label backport_reject_10_x as appropriate.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kmarsden","name":"kmarsden","emailAddress":"kmarsdenderby at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Kathey Marsden","active":true},"created":"2012-03-14T17:19:50.538+0000","updated":"2012-03-14T17:19:50.538+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/13234016","id":"13234016","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mikem","name":"mikem","emailAddress":"mikem_app at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Mike Matrigali","active":true},"body":"temp assigning issue to myself while working on backport.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mikem","name":"mikem","emailAddress":"mikem_app at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Mike Matrigali","active":true},"created":"2012-03-21T01:27:35.926+0000","updated":"2012-03-21T01:27:35.926+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/13234344","id":"13234344","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kmarsden","name":"kmarsden","emailAddress":"kmarsdenderby at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Kathey Marsden","active":true},"body":"This was reported by a user 10.3. Updating affects version","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kmarsden","name":"kmarsden","emailAddress":"kmarsdenderby at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Kathey Marsden","active":true},"created":"2012-03-21T13:23:05.345+0000","updated":"2012-03-21T13:23:05.345+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/13242652","id":"13242652","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mikem","name":"mikem","emailAddress":"mikem_app at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Mike Matrigali","active":true},"body":"i have finished backporting.  backported to 10.7, 10.6, 10.5. 10.4, 10.3.  reassigning original owner.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=mikem","name":"mikem","emailAddress":"mikem_app at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Mike Matrigali","active":true},"created":"2012-03-30T19:14:29.230+0000","updated":"2012-03-30T19:14:29.230+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12428006/comment/13685341","id":"13685341","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"body":"[bulk update] Close all resolved issues that haven't been updated for more than one year.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"created":"2013-06-17T09:19:44.236+0000","updated":"2013-06-17T09:19:44.236+0000"}]},"votes":{"self":"https://issues.apache.org/jira/rest/api/2/issue/DERBY-4275/votes","votes":4,"hasVoted":false},"worklog":{"startAt":0,"maxResults":20,"total":0,"worklogs":[]},"customfield_12311820":"0|i06hy7:"}}